frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -639,24 +696,61 @@ function IndeterminateCheckbox({
 interface EditDocumentKeyProps {
   text: string;
   onEdit: (updatedObject: { column: string }) => void;
+  data: JobMappingFormValues[];
+  schema: string;
+  table: string;
```

Comment:
I suggest actually removing the these props from this component and keeping the EditDocumentKey props much simpler by instead adding an `isDuplicate` callback function that this component can instead call. This will defer the logic to the parent.

I get squeamish whenever I see the `data` being passed around willy nilly because it's generally a heavy object that can be very large. We want to loop over it as minimally as possible. Keeping this component simply by instead having a simply `isDuplicate` function lets us further centralize the data usage.